python/ray/tune/trial.py

```diff
@@ -177,6 +177,28 @@ def has_trainable(trainable_name):
         ray.tune.registry.TRAINABLE_CLASS, trainable_name)
 
 
+def _find_newest_ckpt(ckpt_dir):
+    try:
+        full_paths = [
+            os.path.join(ckpt_dir, fname) for fname in os.listdir(ckpt_dir)
+            if fname.startswith("experiment_state") and fname.endswith(".json")
+        ]
+        with open(max(full_paths), "r") as f:
+            runner_state = json.load(f)
+        log_path = runner_state["checkpoints"][-1]["logdir"]
+        log_paths = [
+            os.path.join(log_path, ckptname, ckptname.replace('_', '-'))
+            for ckptname in os.listdir(log_path)
+            if ckptname.startswith("checkpoint")
+        ]
+        newest_ckpt_path = max(log_paths)
+        logger.info("Find newest checkpoint file {} in {}.".format(
+            newest_ckpt_path, ckpt_dir))
+        return newest_ckpt_path
+    except:
```

Comment:
Can you catch specific exception classes? Namely OSError and JSON reading errors.